mscore/instrwidget.cpp

```diff
@@ -478,6 +480,139 @@ void InstrumentsWidget::genPartList(Score* cs)
+
+void InstrumentsWidget::movePartiturListItems(bool moveUp)
+      {
+            if (_partiturSelectionType == NONE_SELECTED) return;
+            
+            int delta = moveUp ? -1 : 1;
+            QList<QTreeWidgetItem*> originalSelection = partiturList->selectedItems();
+            QList<QTreeWidgetItem*> itemsToMove = QList<QTreeWidgetItem*>(_partiturSelectedItems);
+            PartiturSelectionType selectionType = _partiturSelectionType;
+            partiturList->selectionModel()->clear();
+            
+            QTreeWidgetItem* item;
+            if (selectionType == PARENTS_SELECTED) {
+                  int idx = delta > 0 ? partiturList->topLevelItemCount() - 1 : 0;
+                  for ( ; (item = partiturList->topLevelItem(idx)); idx += -delta) {
+                        if (!itemsToMove.contains(item))
+                              continue;
+                        
+                              bool isExpanded = partiturList->isItemExpanded(item);
+                              int idx = partiturList->indexOfTopLevelItem(item);
+                              int n = partiturList->topLevelItemCount();
+                              // if part not last, move one slot down
+                              if ((idx + delta < n) && (idx + delta >= 0)) {
+                                    QTreeWidgetItem* item1 = partiturList->takeTopLevelItem(idx);
+                                    // Qt looses the QComboBox set into StaffListItem's when they are re-inserted into the tree:
+                                    // get the currently selected staff type of each combo and re-insert
+                                    int numOfStaffListItems = item1->childCount();
+#if (!defined (_MSCVER) && !defined (_MSC_VER))
+                                    int staffIdx[numOfStaffListItems];
+#else
+                                    // MSVC does not support VLA. Replace with std::vector. If profiling determines that the
+                                    //    heap allocation is slow, an optimization might be used.
+                                    std::vector<int> staffIdx(numOfStaffListItems);
+#endif
+                                    int itemIdx;
+                                    for (itemIdx=0; itemIdx < numOfStaffListItems; ++itemIdx)
+                                          staffIdx[itemIdx] = (static_cast<StaffListItem*>(item1->child(itemIdx)))->staffTypeIdx();
+                                    // do not consider hidden ones
+                                    int plusIdx = delta;
+                                    QTreeWidgetItem* nextParent = partiturList->topLevelItem(idx + plusIdx);
+                                    while (nextParent && nextParent->isHidden()) {
+                                          plusIdx += delta;
+                                          nextParent = partiturList->topLevelItem(idx + plusIdx);
+                                    }
```

Comment:
Indentation: closing brackets are indented one tab further than the start of the block, e.g. this should have an extra tab (6 spaces) before it.

Reply:
Thanks, it's very hard to get this to work properly in xcode...